src/OrchardCore.Modules/OrchardCore.Roles/Startup.cs

```diff
@@ -42,6 +42,7 @@ public override void ConfigureServices(IServiceCollection services)
             services.AddRecipeExecutionStep<RolesStep>();
 
             services.AddScoped<IFeatureEventHandler, RoleUpdater>();
+            services.AddScoped<IModularTenantEvents, RoleUpdater>();
```

Comment:
This would create a second instance. You can register `<RoleUpdater>` directly as scoped, then `<IFeatureEventHandler>` and `<IModularTenantEvents>` as `services.AddScoped<...>(sp => sp.GetRequiredService<RoleUpdater>)

Reply:
@sebastienros updated